android/src/main/java/com/polidea/blemulator/bridging/DartMethodCaller.java

```diff
@@ -531,6 +543,278 @@ public void notImplemented() {
+        Map<String, Object> arguments = new HashMap<String, Object>() {{
+            put(ArgumentName.SERVICE_IDENTIFIER, serviceIdentifier);
+            put(ArgumentName.CHARACTERISTIC_UUID, characteristicUUID);
+            put(ArgumentName.DESCRIPTOR_UUID, descriptorUUID);
+            put(ArgumentName.TRANSACTION_ID, transactionId);
+        }};
+
+        dartMethodChannel.invokeMethod(
+                DartMethodName.READ_DESCRIPTOR_FOR_SERVICE,
+                arguments,
+                new MethodChannel.Result() {
+                    @Override
+                    public void success(Object descriptorJsonObject) {
+                        successCallback.onSuccess(descriptorDartValueDecoder.decode((Map<String, Object>) descriptorJsonObject));
+                    }
+
+                    @Override
+                    public void error(String errorCode, String jsonBody, Object irrelevant) {
+                        errorCallback.onError(jsonToBleErrorConverter.bleErrorFromJSON(jsonBody));
+                    }
+
+                    @Override
+                    public void notImplemented() {
+                        Log.e(TAG, "readDescriptorForService not implemented");
+                    }
+                });
+    }
+
+    public void readDescriptorForCharacteristic(final int characteristicIdentifier,
+                                                final String descriptorUUID,
+                                                final String transactionId,
+                                                final OnSuccessCallback<Descriptor> successCallback,
+                                                final OnErrorCallback errorCallback) {
+        Map<String, Object> arguments = new HashMap<String, Object>() {{
+            put(ArgumentName.CHARACTERISTIC_IDENTIFIER, characteristicIdentifier);
+            put(ArgumentName.DESCRIPTOR_UUID, descriptorUUID);
+            put(ArgumentName.TRANSACTION_ID, transactionId);
+        }};
+
+        dartMethodChannel.invokeMethod(
+                DartMethodName.READ_DESCRIPTOR_FOR_CHARACTERISTIC,
+                arguments,
+                new MethodChannel.Result() {
+                    @Override
+                    public void success(Object descriptorJsonObject) {
+                        successCallback.onSuccess(descriptorDartValueDecoder.decode((Map<String, Object>) descriptorJsonObject));
+                    }
+
+                    @Override
+                    public void error(String errorCode, String jsonBody, Object irrelevant) {
+                        errorCallback.onError(jsonToBleErrorConverter.bleErrorFromJSON(jsonBody));
+                    }
+
+                    @Override
+                    public void notImplemented() {
+                        Log.e(TAG, "readDescriptorForCharacteristic not implemented");
+                    }
+                });
+    }
+
+    public void readDescriptor(final int descriptorIdentifier,
```

Comment:
Maybe `readDescriptorForIdentifier(...)` method name would be more in-line with other names? Same goes for `writeDescriptor()`